Mathlib/Tactic/Widget/StringDiagram.lean

```diff
@@ -0,0 +1,286 @@
+/-
+Copyright (c) 2024 Yuma Mizuno. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yuma Mizuno
+-/
+import ProofWidgets.Component.PenroseDiagram
+import Mathlib.Tactic.CategoryTheory.Monoidal
+
+/-!
+# String Diagrams
+
+This file provides tactic/meta infrastructure for displaying string diagrams for morphisms
+in monoidal categories in the infoview.
+
```

Comment:
I would appreciate if you expand the doc a little bit with what a string diagram is and what you exactly implemented.

Can be a very brief/rudimentary, maybe including an almost trivial example, or a reference. Just so that the unfamiliar reader has some starting point.